ada/language/ast.py

```diff
@@ -8209,7 +8296,16 @@ def find_all_references(units=AnalysisUnit.array):
         units.
         """
         dn = Var(Entity.canonical_part)
-        return dn.filter_is_imported_by(units, True).mapcat(
+
+        # If `dn` defines a subprogram which overrides some subprogram P, we
```

Comment:
I took a long time figuring that out so tell me if I'm wrong:

It's OK to take `base` instead of `dn` because every derived instance necessarily "withes"  the unit in which the parent is defined, and transitively up to the base. This wouldn't be true if `transitive` wasn't set to true below.

If that's correct, that might warrant an additional comment :)

Reply:
Yes that's it.
Done.